iml-systemd/src/lib.rs

```diff
@@ -103,7 +107,7 @@ async fn get_unit_states(unit_name: &str) -> Result<(UnitFileState, ActiveState)
 /// # Arguments
 ///
 /// * `unit_name` - The unit to start
-pub async fn start_unit(unit_name: String) -> Result<(), ImlAgentError> {
+pub async fn start_unit_with_time(unit_name: String, time: u32) -> Result<(), SystemdError> {
```

Comment:
```suggestion
pub async fn start_unit_and_wait(unit_name: String, time: u32) -> Result<(), SystemdError> {
```

Would better reflect intent I think.

Reply:
sure 👍